gitlab/resource_gitlab_tag_protection.go

```diff
@@ -117,3 +133,21 @@ func projectAndTagFromID(id string) (string, string, error) {
 	}
 	return project, tag, err
 }
+
+func resourceGitlabTagProtectionImporter(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	s := strings.Split(d.Id(), ":")
+	if len(s) != 2 {
+		d.SetId("")
+		return nil, fmt.Errorf("invalid tag protection import format; expected '{project_id}:{tag_name}'")
+	}
+	project, tag := s[0], s[1]
+
+	d.SetId(buildTwoPartID(&project, &tag))
+	if err := d.Set("project", project); err != nil {
+		return nil, err
+	}
+	if err := d.Set("tag", tag); err != nil {
+		return nil, err
+	}
+	return []*schema.ResourceData{d}, nil
+}
```

Comment:
If you accept my `schema.ImportStatePassthrough` suggestion above, you can also delete this function. The reason is that the `Read` function only uses the `id` attribute, and it also sets the `project` and `tag` attributes for you. Since `Read` is always called during an import, this importer function is redundant.
```suggestion
```